### ISSUE TITLE
anchor tag stores block element in html4

### ISSUE THREAD

**Issue by khoabui1412** [htacg/tidy-html5#956]
Hi,
I have input file:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" 
"http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8">
<title>p moved out of a #169</title>
</head>
<body>
<a href="l1">
<div>one</div>
</a>
</body>
</html>
```
After parsing, output will be:
```
...
<a href="l1"/>
<div>one</div>
...
```
It's related to issue [link](https://github.com/htacg/tidy-html5/issues/169), but now most of browsers treat it as in html5 though it's doctype is html4. So should we do the same behavior of those browsers?

**Comment by balthisar** [comment]
I agree that upon testing, modern browsers are accepting block level elements nested within anchors. Rather than breaking tidy's existing behavior, though, I wonder if we'd be better adding a new option such as `anchor-nests-blocks` or similar, which would only have effect when not using the HTML5 parser?

If that's the case, let me know if I should mark this a a feature request.

**Comment by khoabui1412** [comment]
yes @balthisar , there're some other related cases in `TY_(AdjustTags)` function, so can consider adding one option such as `html5-parser` for all of those cases
